pkg/config/config.go

```diff
@@ -1,26 +1,75 @@
 package config
 
+import (
+	"fmt"
+
+	"github.com/caarlos0/env"
+	"github.com/golang/glog"
+)
+
 type DaemonConfig struct {
-	Plugin         string `json:"plugin"`          // Subnet manager plugin name
-	PeriodicUpdate int    `json:"periodic_update"` // Interval between every check for the added and deleted pods
-	GuidPool       GuidPoolConfig
-	SubnetManager  SubnetManagerPluginConfig
+	Plugin         string                    `json:"plugin" env:"PLUGIN"`                   // Subnet manager plugin name
```

Comment:
thats the subnet manager plugin right ?
IMO this needs to be partitioned under SubnetManagerPluginConfig
name can be derrived explicitly or implicitly depends on the example i proposed

Reply:
moved